datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -673,7 +673,16 @@ impl GroupedHashAggregateStream {
         let spillfile = self.runtime.disk_manager.create_tmp_file("HashAggSpill")?;
         let mut writer = IPCWriter::new(spillfile.path(), &emit.schema())?;
         // TODO: slice large `sorted` and write to multiple files in parallel
-        writer.write(&sorted)?;
+        let mut offset = 0;
+        let total_rows = sorted.num_rows();
+
+        while offset < total_rows {
+            let length = std::cmp::min(total_rows - offset, self.batch_size);
+            let batch = sorted.slice(offset, length);
+            offset += batch.num_rows();
+            writer.write(&batch)?;
+        }
+
```

Comment:
Is it possible to write in parallel? So that this becomes less blocking.

Additional improvement would be that chunking before sorting. I remember the discussion was `sorted` keeps a copy in memory. We can slice `emit` before `sort_batch` and sort right before writing.

Reply:
I don't see big benefit of writing in parallel, but we can give it a try at the later date.
Will try to implement some of the alternatives we discuss, at the moment addressing low hanging fruits.

Reply:
Got it.
Would you mind explaining how reading smaller files reduces memory usage? I thought we are streaming when reading back for merging. Also, when merge-sort is going on, we need to open all files anyway? Just making sure the original issue statement in #8003 

Additionally, I think we have to create a new writer otherwise, we are keep appending to the same temp file? So we end up having the same file size?

Reply:
Sorry for the confusion, not smaller files, smaller batches. We still keep one file per spill method invocation

Previous implementation would just have one batch per file, this change would introduce more than one batch per file. So file size should be similar size to previous implantation, probably even a bit bigger, but that same file will have more than one batch in it.
 
Streaming merge will open same number of files like before, but it will load smaller batches into memory (if we have one batch per file it would mean whole file would be loaded).

Reply:
Thank you for explaining!